### ISSUE TITLE
NA OFI: error following bulk transfer: fi_cq_readerr got err: 5 (Input/output error)

### ISSUE THREAD

**Issue by yulujia** [mercury-hpc/mercury#313]
While running the IOR test in DAOS over the sockets provider, at some point one of the 4 servers starts to get an error like this repeatedly, I checked the listening tcp ports by running:
```
 ss -l | grep LISTEN
``` 
the ports used by server are all there. @soumagne any idea why might be causing this?

```
09/26-09:53:03.96 [138749/138767] object DBUG src/object/srv_obj.c:216 ds_bulk_transfer() bulk_op:105 sgl_nr1
09/26-09:53:03.96 [138749/138767] common DBUG src/common/misc.c:44 daos_sgl_init() alloc(calloc) 'sgl->sg_iovs': 24 at 0x7f87e7384820.
09/26-09:53:03.96 [138749/138767] rpc  DBUG src/cart/crt_rpc.c:472 crt_req_addref(0x7f87e7383f60)  [opc=0x401000b xid=0x15a4 rank:tag=3:7] addref to 2.
09/26-09:53:03.96 [138749/138767] hg   DBUG src/cart/crt_hg.c:1622 crt_hg_bulk_transfer() alloc(calloc) 'bulk_cbinfo': 24 at 0x7f87e73848b0.
09/26-09:53:03.96 [138749/138767] hg   DBUG src/cart/crt_hg.c:1625 crt_hg_bulk_transfer() alloc(calloc) 'bulk_desc_dup': 56 at 0x7f87e73848d0.
09/26-09:53:03.96 [138749/138767] common DBUG src/common/misc.c:67 daos_sgl_fini() free 'sgl->sg_iovs' at 0x7f87e7384820.
09/26-09:53:03.96 [138749/138767] hg   ERR  # NA -- Error -- mercury/src/na/na_ofi.c:2770
 # na_ofi_cq_read(): fi_cq_readerr got err: 5(Input/output error), prov_errno: -5(Input/output error).
```

**Comment by soumagne** [comment]
Hmm not sure, could you maybe set FI_LOG_LEVEL to debug ? this way we may have a chance to see what's going on.

**Comment by yulujia** [comment]
If I set FI_LOG_LEVEL to debug the log file grows to gigabytes quickly. Just found a way to deal with that. I will get some logs today.

**Comment by yulujia** [comment]
@soumagne  I set FI_LOG_LEVEL to debug, the ofi log doesn't containing any error, only debug messages. The ofi log is a few hundred MB, not sure how to upload it here.

**Comment by soumagne** [comment]
hmm ok well do you think you could set a breakpoint at that error line and try to get a trace? without more debug information it will be hard to know what the problem is...

**Comment by yulujia** [comment]
I can get a trace when we hit the error on mercury/src/na/na_ofi.c:2770, but would that be useful? When we hit the error mercury is processing the event from the completion queue, the error really happened at an earlier time.

**Comment by soumagne** [comment]
Ah right, I think it would be useful if you can try to grep for errors in the OFI log... maybe FI_LOG_LEVEL=warn would be enough?

**Comment by yulujia** [comment]
I managed to get the logs from the server process that has the error.

full libfabric log with FI_LOG_LEVEL=debug: https://app.box.com/s/kejfbpb9oa5bppztp5plqeo94oy49w7t

error message from the application log:
```
> awk '/Error/ || /error/' daos_server.log

10/09-13:59:50.77 boro-38 DAOS[105642/105661] hg   ERR  # NA -- Error -- mercury/src/na/na_ofi.c:2770
 # na_ofi_cq_read(): fi_cq_readerr got err: 5(Input/output error), prov_errno: -5(Input/output error).
10/09-13:59:50.77 boro-38 DAOS[105642/105661] hg   ERR  # NA -- Error -- mercury/src/na/na_ofi.c:2770
 # na_ofi_cq_read(): fi_cq_readerr got err: 5(Input/output error), prov_errno: -5(Input/output error).
10/09-14:00:20.77 boro-38 DAOS[105642/105661] hg   ERR  # NA -- Error -- mercury/src/na/na_ofi.c:2770
 # na_ofi_cq_read(): fi_cq_readerr got err: 5(Input/output error), prov_errno: -5(Input/output error).

```

**Comment by soumagne** [comment]
We now ignore this type of events, fixed in eff8652eb0a22c4acec63ec3b47321a7d7547f2e
